src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -385,15 +386,16 @@ void OTARequestor::ApplyUpdate()
     ConnectToProvider(kApplyUpdate);
 }
 
-void OTARequestor::NotifyUpdateApplied(uint32_t version)
+void OTARequestor::NotifyUpdateApplied()
 {
-    // New version is executing so update where applicable
-    VerifyOrReturn(Basic::Attributes::SoftwareVersion::Set(kRootEndpointId, version) == EMBER_ZCL_STATUS_SUCCESS);
+    // New version is executing so update cached version
+    uint32_t version;
+    VerifyOrReturn(BasicClusterServerGetSoftwareVersionAttribute(kRootEndpointId, version) == EMBER_ZCL_STATUS_SUCCESS);
     mCurrentVersion = version;
 
     // Log the VersionApplied event
     uint16_t productId;
-    VerifyOrReturn(Basic::Attributes::ProductID::Get(kRootEndpointId, &productId) == EMBER_ZCL_STATUS_SUCCESS);
+    VerifyOrReturn(BasicClusterServerGetProductIdAttribute(kRootEndpointId, productId) == EMBER_ZCL_STATUS_SUCCESS);
```

Comment:
The previous API was more correct

Reply:
No, because `Attributes::ProductID::Get` _requires_ the value to be stored in RAM or "external" via app callback right now.  So it would fail to work correctly if basic information switched to using an AttributeAccessInterface for this attribute.

Abstracting away that knowledge of exactly how basic information is gets the attribute value is the point of the new API.